zer0-os/zOS
# feat: map creator id to current users zero user id

### What does this do?
- sets the admin `creatorId` to the senders zero user id. This is specifically for creating room events, where a sender id is available in the event data.

### Why are we making this change?
- this ensures that when there is admin data, specifically a `creatorId`, the id will be mapped to the senders zero user id. This is required to ensure correct logic for admin messages works (using zero user ids).

### How do I test this?
- For a quick test, you could listen for `EventType.RoomCreate` in `processRawEventsToMessages` when returning `mapEventToAdminMessage` and add a console.log for the channel in the saga. This will log data as per the screenshot below where you will see the `creatorId` as the senders zero user id.

### Key decisions and Risk Assessment:
  #### Things to consider:
  1. How will this affect security?
  1. How will this affect performance?
  1. Does this change any APIs?

<img width="1229" alt="Screenshot 2023-11-13 at 12 48 22" src="https://github.com/zer0-os/zOS/assets/39112648/5ffcc909-0445-4d5d-911b-7499f05883f6">


Note: here we see two admin messages due to `processRawEventsToMessages` processing both the custom event and the room create event. This is the next part of the admin messages task to implement.

The logic will need to be enhanced, but for testing purposes the logic will work as expected.
<img width="1120" alt="Screenshot 2023-11-13 at 14 27 27" src="https://github.com/zer0-os/zOS/assets/39112648/6fe82cdd-7a7d-40f5-9098-59b3febc0899">
